lib/logflare/backends.ex

```diff
@@ -478,21 +479,62 @@ defmodule Logflare.Backends do
     end
   end
 
+  def push_recent_events(%Source{}, []), do: :ok
+
+  def push_recent_events(%Source{id: source_id}, log_events) do
+    existing = :ets.lookup(:recent_events, source_id)
+
+    :ets.delete(:recent_events, source_id)
```

Comment:
This seems hairy. We're going to have many processes doing this at once.

Reply:
My feeling is that we're going to want to do only inserts here and then a a janitor process doing the deletes in the background, similar to how Cachex does maintains ets tables.

But, since this powers only the initial logs list view we can punt if the perf is there.

Reply:
Yeah so for very busy sources the snapshot of the "last 100" you get when you load the page is not really going to be the last 100 with  many processes doing this or even doing `:ets.take`.

But again, might be close enough.

Reply:
Actually @Ziinc if the goal here is to reduce GC, then we need to do the janitor process.

Because every time we select data from ETS we're copying it and generating double the garbage we need to. 

If we have a janitor process and delete without returning any data than I think we'd significantly reduce copying.